pdns/dnsdist-idstate.hh

```diff
@@ -157,73 +157,25 @@ struct IDState
   IDState(const IDState& orig) = delete;
   IDState(IDState&& rhs)
   {
-    if (rhs.isInUse()) {
-      throw std::runtime_error("Trying to move an in-use IDState");
-    }
-
-#ifdef __SANITIZE_THREAD__
+    inUse.store(rhs.inUse.load());
     age.store(rhs.age.load());
-#else
-    age = rhs.age;
-#endif
     internal = std::move(rhs.internal);
   }
 
   IDState& operator=(IDState&& rhs)
   {
-    if (isInUse()) {
-      throw std::runtime_error("Trying to overwrite an in-use IDState");
-    }
-
-    if (rhs.isInUse()) {
-      throw std::runtime_error("Trying to move an in-use IDState");
-    }
-#ifdef __SANITIZE_THREAD__
+    inUse.store(rhs.inUse.load());
     age.store(rhs.age.load());
-#else
-    age = rhs.age;
-#endif
-
     internal = std::move(rhs.internal);
```

Comment:
clang-tidy prefers to move this assignment to a member init and to mark this op noexcept